modules/nf-core/nanocomp/main.nf

```diff
@@ -0,0 +1,35 @@
+process NANOCOMP {
+    label 'process_medium'
+
+    conda "bioconda:nanocomp=1.21.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/nanocomp:1.21.0--pyhdfd78af_0':
+        'quay.io/biocontainers/nanocomp:1.21.0--pyhdfd78af_0' }"
+
+    input:
+    val filetype
+    path filelist
+
+    output:
+    path "*.html", emit: html_nanocomp_output
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    """
+    if ! [[ $filetype = @(fastq|fasta|summary|bam) ]]; then echo "Input should only be fastq, fasta, summary or bam type, but is $filetype" && exit 1; fi
+    NanoComp \\
+        --$filetype $filelist \\
```

Comment:
Can you derive the file type from the file name?

Reply:
I would not like to do that, honestly, since the ending of the file name can differ (fasta, fa, maybe even more, both additionally gzipped, just for fasta), so it would probably be tons of queries for that. Is there maybe a good example which already does that? :)

Reply:
It's mainly because we don't really like value inputs that could be derived from the data. Here's an example for an output extension: https://github.com/nf-core/modules/blob/60035b765c1c9feb18f27443ee0439a1eb94084d/modules/nf-core/bcftools/sort/main.nf#L23-L27
You can do something similar here with input extensions